api/loader/fileloader.go

```diff
@@ -302,7 +307,7 @@ func (fl *fileLoader) errIfRepoCycle(newRepoSpec *git.RepoSpec) error {
 // else an error. Relative paths are taken relative
 // to the root.
 func (fl *fileLoader) Load(path string) ([]byte, error) {
-	if u, err := url.Parse(path); err == nil && (u.Scheme == "http" || u.Scheme == "https") {
+	if HasRemoteFileScheme(path) {
```

Comment:
My interpretation of the lint error:
I put the condition for the if statement into a function. The linter ran on the if statement as a consequence and found deeply nested if statements in the old code.